cmake/ChooseCudaCC.cmake

```diff
@@ -0,0 +1,89 @@
+#
+# after returning from this function, do not forget to call the following:
+#    set(RESULT_NAME ${RESULT_NAME} CACHE STRING "CUDA CC versions to compile")
+# replacing your own variable for RESULT_NAME
+#
+# We assume that MINCC default to 20
+# We assume that MINCUDAVERSION defaults to 7.0
+#
+function(chooseCudaCC RESULT_NAME MINCC MINCUDAVERSION)
+  if(NOT DEFINED MINCC)
+    set(MINCC 20)
+  endif()
+  if(NOT DEFINED MINCUDAVERSION)
+    set(MINCUDAVERSION 7.0)
+  endif()
+
+  find_package(CUDA ${MINCUDAVERSION} REQUIRED)
+
+  if(NOT CUDA_FOUND)
+    message(FATAL_ERROR "Could not find CUDA >= ${MINCUDAVERSION}")
+  endif()
+
+  #
+  # Create a list of possible CCs for each host processor.
+  # This may require tuning: CUDA cards exist in AIX machines with POWER CPUs,
+  # it is possible that non-Tegra ARM systems exist as well.
+  # For now, this is my best guess.
+  #
+  if((CMAKE_SYSTEM_PROCESSOR STREQUAL "i686") OR (CMAKE_SYSTEM_PROCESSOR STREQUAL "x86_64"))
```

Comment:
apparently we need to add `AMD64` for windoze? https://ci.appveyor.com/project/AliceVision/popsift/builds/31043728#L154

Reply:
also we can simplify:
```cmake
set(SUPPORTED_PROCESSORS i686;x86_64;AMD64)
if(CMAKE_SYSTEM_PROCESSOR IN_LIST SUPPORTED_PROCESSORS)
```

Reply:
I guess we could use two variables TEGRA_SUPPORTED_PROCESSORS and OTHER_SUPPORTED_PROCESSORS?

Reply:
yes, it was more to suggest to use `IN_LIST` that simplify the condition

Reply:
I'll try that. Because of the string-list identity in CMake, we would need to set "i686;x86_64;AMD64", right?

Reply:
yes